### PR TITLE
Replace icons with codicons

### LINKED ISSUE #6115: Update VSCode Icons
**Description**

The repository [vscode-icons](https://github.com/microsoft/vscode-icons) has recently been created where we can re-use the same icons present in VSCode. Is it something we'd like to align to? If we use the outdated icons it's quite possible we will get a mismatch in our own applications when using plugins (which bring the new icons in their views).

### PR DESCRIPTION
**Pull request by msujew**
#### What it does

Closes #6115

Does a lot of css work:

* Replaces all internally used icons (fontawesome, custom svg files) with codicons
* Adds a new utility method for codicons, namely `codicon` and `codiconArray`.
* Aligns button-icon css with vscode: Buttons in toolbars, etc. are now highlighted with a backdrop

I decided to leave the old svg icons in the codebase for backwards compatibility. I also couldn't notice any breaking change, but please keep an eye out for them.

#### How to test

Before starting to test I would recommend to start a second Theia instance to compare visual changes. The following testing steps are only the most important parts that have changed. As every icon in the app has changed, please refer to the code to gather a list of everything that has changed. Also, change the icon theme to `seti` at appropriate times to check for tab-bar items, etc.

* Check toolbar icons, every icon should show a backdrop on hover.
* Check tabbar close icons, every close icon should show a backdrop on hover.
* Check status bar icons
* Check notification widget icons
* Check debug view widget icons
* Check scm view widget icons
* Check problems view widget icons
* Check extensions view widget icons

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by vince-fugnitto — COMMENTED** [review]
@msujew I believe we should also update the widget icons themselves (ex: `explorer`, `scm`, ...), any reason we wanted to omit it initially?

**Comment by msujew** [comment]
> any reason we wanted to omit it initially?

Actually no, I just stopped thinking about it, once I noticed that the explorer icon is not in the codicon collection. But I just saw that every other icon is in there. So I'll replace those. What is your opinion on replacing the default folder/file icon in the explorer? Currently it still uses the fontawesome icons.

**Comment by vince-fugnitto** [comment]
> Actually no, I just stopped thinking about it, once I noticed that the explorer icon is not in the codicon collection. But I just saw that every other icon is in there. So I'll replace those. What is your opinion on replacing the default folder/file icon in the explorer? Currently it still uses the fontawesome icons.

I think the idea for consistency with vscode and what plugins use we should aim to update all icons (including font-awesome icons) to their codicon counterparts. This includes widgets when docked to sidepanels. It will create a uniform look and feel across the framework.

File: packages/callhierarchy/src/browser/callhierarchy-tree/callhierarchy-tree-widget.tsx [hunk 1]
```diff
@@ -177,7 +177,8 @@ export class CallHierarchyTreeWidget extends TreeWidget {
 
     private doOpenEditor(uri: DocumentUri, range: Range, keepFocus: boolean): void {
         this.editorManager.open(
-            new URI(uri), {
+            new URI(uri),
+            {
```
Comment (vince-fugnitto) [inline packages/callhierarchy/src/browser/callhierarchy-tree/callhierarchy-tree-widget.tsx:181]: minor: unrelated change.

File: packages/core/src/browser/label-provider.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ import { ResourceLabelFormatter, ResourceLabelFormatting } from '../common/label
 /**
  * @internal don't export it, use `LabelProvider.folderIcon` instead.
  */
-const DEFAULT_FOLDER_ICON = 'fa fa-folder';
+const DEFAULT_FOLDER_ICON = '';
```
Comment (vince-fugnitto) [inline packages/core/src/browser/label-provider.ts:30]: Can you confirm why we want the default folder icon to be non-existent?

File: packages/core/src/browser/progress-status-bar-item.ts [hunk 3]
```diff
@@ -73,7 +73,7 @@ export class ProgressStatusBarItem implements ProgressClient {
             this.statusBar.removeElement(this.id);
             return;
         }
-        const text = `$(refresh~spin) ${message}`;
+        const text = `$(codicon-sync~spin) ${message}`;
```
Comment (vince-fugnitto) [inline packages/core/src/browser/progress-status-bar-item.ts:76]: This reminds me of the following issue https://github.com/eclipse-theia/theia/issues/9768, perhaps it's something we can easily implement in the future.

File: packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts [hunk 4]
```diff
@@ -60,7 +60,7 @@ export class ModalNotification extends AbstractDialog<string | undefined> {
         const iconContainer = messageNode.appendChild(document.createElement('div'));
         iconContainer.classList.add(ICON);
         const iconElement = iconContainer.appendChild(document.createElement('i'));
-        iconElement.classList.add('fa', this.toIconClass(messageType), 'fa-fw', messageType.toString());
+        iconElement.classList.add(... this.toIconClass(messageType), 'fa-fw', messageType.toString());
```
Comment (vince-fugnitto) [inline packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts:63]: nitpick: the `...` formatting looks a bit off.

File: packages/core/src/browser/label-provider.ts (hunk 2 above)
Comment (msujew) [reply packages/core/src/browser/label-provider.ts:30]: Because that's the way it works in vscode (default icon theme). I wanted to align with that.

**Review by vince-fugnitto — COMMENTED** [review]
If we previously had layout data saved (ex: local-storage), then starting the app may cause icons to be non-existent:

image


We may want to introduce a layout-migration:

https://github.com/eclipse-theia/theia/blob/1dba01446fd7aef464bcaef5f853d6f595f96bbd/packages/core/src/browser/shell/application-shell.ts#L51-L62

https://github.com/eclipse-theia/theia/blob/1dba01446fd7aef464bcaef5f853d6f595f96bbd/packages/core/src/browser/shell/shell-layout-restorer.ts#L87-L110

File: packages/core/src/browser/label-provider.ts (hunk 2 above)
Comment (vince-fugnitto) [reply packages/core/src/browser/label-provider.ts:30]: I believe this is for our custom `file-icons` theia theme, not the vscode theme:

_before_:

image

_after_:

image


I don't think we should break, if users want the vscode experience they would use `seti` (default in vscode):

image

Comment (msujew) [reply packages/core/src/browser/label-provider.ts:30]: I see. Would you recommend to keep the old `fa` icon then? Or rather replace it with the codicon as well?

Comment (vince-fugnitto) [reply packages/core/src/browser/label-provider.ts:30]: Up to you, I think `codicon folder` would look nice :+1:, and perhaps in the future we can think of deprecating font-awesome in the framework altogther.

**Comment by msujew** [comment]
> We may want to introduce a layout-migration

Alright, done. It's a bit crude, but I found the string replacement method to be sufficient.

File: packages/core/src/browser/label-provider.ts (hunk 2 above)
Comment (msujew) [reply packages/core/src/browser/label-provider.ts]: Alright, replaced them with codicons 👍

**Review by vince-fugnitto — COMMENTED** [review]
@msujew do you mind rebasing the pull-request when you get a chance? I'd be happy to take a look at it afterwards :)

**Review by vince-fugnitto — COMMENTED** [review]
I really love the new icons and behavior for toolbar items :+1: 
I noticed two things when testing:
- I believe the bottom sidebar (settings icon) may need to be adjusted to have the same `font-size` as the rest of the sidebar
- I believe the search-in-workspace widget needs to be updated to take into account the new chevron (left is the pr, and right is master), the input fields are no longer aligned:

siw-diff

- dirty editors are no longer marked as dirty:

image

**Comment by msujew** [comment]
@vince-fugnitto Thanks for the feedback, I think I addressed everything 👍

**Review by vince-fugnitto — COMMENTED** [review]
I'd be fine to merge and fix issues throughout the month if necessary as I do not want to hold back the pull-request :+1: 


I still notice some UI issues, perhaps one of the bigger ones (that may be caused by a bigger issue) is that the plugin-views sometimes do not display their proper icons:

image

I also noticed that the alignment of toolbar items contributed by plugins are not necessarily aligned properly as well (compared to builtin views like the explorer):

image


A very minor bug is how the scm arrow is not aligned properly (for `CHANGES`) compared to other trees:

image

**Comment by msujew** [comment]
> I still notice some UI issues, perhaps one of the bigger ones (that may be caused by a bigger issue) is that the plugin-views sometimes do not display their proper icons

This seems to be unrelated to my changes, see the current `master`:

image

I noticed that the `instanceof` check in the plugin host fails:

https://github.com/eclipse-theia/theia/blob/7442150b335a9280a314dbf30f6a9227305b6b04/packages/plugin-ext/src/plugin/tree/tree-views.ts#L372

I assume this is due to the ES2017 upgrade and the `@es5Compatibility` annotation (#9436). `instanceof` checks are not a valid operation on es5 classes as far as I can tell.

I'll add an `is` check for `ThemeIcon` users instead.

**Comment by vince-fugnitto** [comment]
@msujew sorry I had thought I confirmed against master that it was a regression (turns out the code I confirmed against was older) but you're right that its not related to your changes and we can omit the fix for now :+1:

**Review by vince-fugnitto — APPROVED** [review]
The changes look good to me :+1: 

I think there is a leftover file:

image

**Comment by msujew** [comment]
I'll merge this Tuesday or Wednesday, if no one else has any issues with my changes.
